crates/rune-macros/src/context.rs

```diff
@@ -358,6 +378,67 @@ impl Context {
                             }
                         },
                     });
+                } else if meta.path == GET_REF {
+                    attr.field = true;
+                    attr.protocols.push(FieldProtocol {
+                        custom: self.parse_field_custom(meta.input)?,
+                        generate: |g| {
+                            let Generate {
+                                ty,
+                                target,
+                                ..
+                            } = g;
+
+                            match target {
+                                GenerateTarget::Named { field_ident, field_name } => {
+                                    let protocol = g.tokens.protocol(PROTOCOL_GET);
+
+                                    match ty {
+                                        syn::Type::Reference(TypeReference{ mutability: None, ..}) => {
+                                            quote_spanned! { g.field.span() =>
+                                                module.field_function(#protocol, #field_name, |s: &Self| {
+                                                    let s = Shared::new(AnyObj::new_projection(s.#field_ident));
```

Comment:
Hm, it seems like these references are leaking?

To perform a projection like this, we need to access the `Shared<AnyObj>` which is invalidated at the end of the vm call. If we're constructing a completely new `Shared` here, it is no longer part of the same argument which is invalidated by [`GuardedArgs`](https://docs.rs/rune/latest/rune/runtime/trait.GuardedArgs.html).

The reason passed in references is *safe*, is because the `Shared<T>` object that is produced from it is invalidated through `GuardedArgs`, so we must make sure any projections do the same.

Reply:
A simple test you can try here is to return a reference (held by a `Value`) from the `Vm::call`. This should not be allowed, because it allows the caller to extend the lifetime beyond the object from where it was projected.